src/lib.rs

```diff
@@ -857,10 +857,33 @@ impl BumpFeeTxBuilder {
     }
 }
 
-fn generate_mnemonic(word_count: WordCount) -> Result<String, BdkError> {
-    let mnemonic: GeneratedKey<_, BareCtx> =
-        Mnemonic::generate((word_count, Language::English)).unwrap();
-    Ok(mnemonic.to_string())
+struct Mnemonic {
+    internal: BdkMnemonic,
+}
+
+impl Mnemonic {
+    fn new(word_count: WordCount) -> Self {
+        let generated_key: GeneratedKey<_, BareCtx> =
+            BdkMnemonic::generate((word_count, Language::English)).unwrap();
+        let mnemonic = BdkMnemonic::parse_in(Language::English, generated_key.to_string()).unwrap();
```

Comment:
I can't quite figure out if there is a better way to create this mnemonic but for some reason these 3 lines look and feel a bit weird/complex to me, when there should be some sort of `Mnemonic::create()` method that does it all in one go. Instead we have to use the `Mnemonic::generate` which actually returns a `GeneratedKey` _which has a mnemonic inside_, and then we feed that  mnemonic back to the `Mnemonic` struct?

But I'm looking at the docs and for some reason I can't find a better/cleaner way. But as is, doesn't it feel like we're not quite using the API as intended? Surely this is a very convoluted way to get at the idea of "creating a mnemonic".

Reply:
I think this might be due to the fact that bdk does not turn on the `rand` feature on the bip39 crate. From the docs and some of their examples, I gather that the "standard" way to do what we're trying to do here would be to use the `Mnemonic::generate()` or `Mnemonic::generate_in()` methods ([docs here](https://docs.rs/bip39/1.0.1/src/bip39/lib.rs.html#275-295)). This is where it would just use the RNG and create the mnemonic directly for you. 

Going through the `GeneratedKey<_ BareCtx> struct is a workaround that I guess, but I'm not as clear as to what this type does/purpose.

Reply:
I've found a workaround for this
It is to use the `Mnemonic::from_entropy_in()` ([docs here](https://docs.rs/bip39/1.0.1/src/bip39/lib.rs.html#180)).  And the method would look like this :
```rust
fn new(word_count: WordCount) -> Self {
    let mut entropy = [0u8; 32];
    thread_rng().fill(&mut entropy[..]);
    let entropy = &entropy.as_ref()[..(word_count as usize / 8)];
    let mnemonic = BdkMnemonic::from_entropy_in(Language::English, entropy).unwrap();
    Mnemonic { internal: mnemonic }
}
```

Here we cannot use `Mnemonic::generate_in_with()` ([docs here](https://docs.rs/bip39/1.0.1/src/bip39/lib.rs.html#245-261)) because `bip39::Mnemonic` only supports 12, 18 & 24 words. While BDK supports 12, 15, 18, 21 & 24 words. So we can implement it the same way it's done in BDK and have support for 12, 15, 18, 21 & 24 words.

Reply:
Good research, even though it's a little convoluted it looks like the current approach is the only way to get what we need without copy/pasting code. I think you're right @thunderbiscuit we need to push for  bitcoindevkit/bdk#644 and make sure it provides a better API to do what we need here.

Reply:
So do we continue as it is right now, and cleanup the API with different PR once [bitcoindevkit/bdk#644](https://github.com/bitcoindevkit/bdk/pull/644) gets merged?
Or should we wait for it to get merged?

Reply:
I suggest we go ahead with what you currently have. Add the docstrings and it's a go from me.

Reply:
Updated the PR according to suggestions. Resolving Conversation